packages/starknet-snap/test/src/executeTxn.test.ts

```diff
@@ -58,93 +79,199 @@ describe('Test function: executeTxn', function () {
     apiParams.requestParams = requestObject;
   });
 
-  it('should executeTxn correctly', async function () {
-    const stub = sandbox.stub(utils, 'executeTxn').resolves({
-      transaction_hash: 'transaction_hash',
+  describe('when account is not deployed', () => {
+    let createAccountStub;
+    beforeEach(function () {
+      createAccountStub = sandbox.stub(createAccountUtils, 'createAccount').resolvesThis();
     });
-    const result = await executeTxn(apiParams);
-    const { privateKey } = await utils.getKeysFromAddress(
-      apiParams.keyDeriver,
-      STARKNET_MAINNET_NETWORK,
-      state,
-      account1.address,
-    );
-
-    expect(result).to.eql({
-      transaction_hash: 'transaction_hash',
-    });
-    expect(stub).to.have.been.calledOnce;
-    expect(stub).to.have.been.calledWith(
-      STARKNET_MAINNET_NETWORK,
-      account1.address,
-      privateKey,
-      {
-        entrypoint: 'transfer',
-        calldata: ['0', '0', '0'],
-        contractAddress: createAccountProxyTxn.contractAddress,
-      },
-      undefined,
-      { maxFee: 100 },
-    );
-  });
-
-  it('should executeTxn multiple', async function () {
-    const stub = sandbox.stub(utils, 'executeTxn').resolves({
-      transaction_hash: 'transaction_hash',
+    afterEach(function () {
+      expect(createAccountStub).to.have.been.calledOnceWith(sinon.match.any, true, true);
```

Comment:
expect statement should not put in afterEach

we usually put some reset statement in afterEach

for determine statement, we put it inside the test, else, it is very hard to follow

and 

actually in MM style, they dont prefer sub test flow
Not:
when xxxx
    when yyyy
       it zzzz

Prefer:
when xxxx and yyyy
   it zzzz

i knew that some testing script is already using sub test flow, but would you mind to change it back

other than that, everything is good to go

Reply:
ok let me roll back this change.